salesagility/SuiteCRM
# solve Issue#9629 (Smarty plugin function "Sugarvar" generates tpl with SyntaxError)

## Description
SuiteCRM "crashes" (http error 500) due to SyntaxError in generated tpl files

**name of generated tpl:**  SearchForm_advanced.tpl

**files used for generating:**

- SearchFormGeneric.tpl
- SearchView.tpl 


**how it affects me:**

1. auto-generated "advanced_search" blocks in Leads, Accounts... listviews
2. click on listview column for sorting
3. see error 500/Smarty Error-SyntaxError in apache log
4. displayed listview page stays empty


**Environment**
- SuiteCRM Version used: 7.12.5


**why does this happen:**
because _function.sugarvar.php_ generates a string, which is used in _SearchView.tpl_ as a variable-field-name, but generated variable-field-name is invalid

**why is the generated filed-name invalid:**
because _$smarty->get_template_vars($params['memberName'])_ in _function.sugarvar.php_ does return empty string/nothing
and the returned value is used unvalidated when generating the field-name

**how did i fix it:**
validate if $member has value
if not, do something to fallback

## Motivation and Context
because SuiteCrm crashes

## How To Test This
1. use the "Quick Repair and Rebuild" function of SuiteCRM
2. make sure that "cache/themes/SuiteP/modules" is empty (or nonexistent anymore)
3. go to one of the ListViews (Leads,Accounts...) in CRM 
4. new directories will be autogenerated in "cache/themes/SuiteP/modules" with tpl files

## Types of changes
- [ x ] Bug fix (non-breaking change which fixes an issue)

### Final checklist
- [ x ] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ x ] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.
